modules/govuk/manifests/apps/asset_manager.pp

```diff
@@ -88,7 +88,40 @@
       location ~ /raw/(.*) {
         internal;
         alias /var/apps/asset-manager/uploads/assets/$1;
-      }',
+      }
+
+      location = /cloud-storage-proxy/ {
+        # Prevent requests to this location from outside nginx
+        internal;
+
+        # Extract download host and URL from upstream HTTP headers
+        set $cloud_storage_host $upstream_http_x_cloud_storage_host;
+        set $download_url $upstream_http_x_cloud_storage_url;
```

Comment:
I've updated the nginx config to send the S3 URL as part of the `X-Accel-Redirect` header. I think the problems I ran into previously were due to my lack of nginx knowledge!